lib/src/wallet/keypair.dart

```diff
@@ -32,7 +33,7 @@ class KeyPair {
   }
 
   Future<List<int>> sign(List<int> message) async {
-    return Crypto.sign(message, privateKey, (await getPublicKey()));
+    return Crypto.sign(Crypto.digest(message), privateKey, (await getPublicKey()));
```

Comment:
Why this change?

Reply:
See the last paragraph in the https://github.com/hypercore-one/znn_sdk_dart/issues/2

An alternative is to pass a transaction interface containing both the source and the hash data as properties?

Reply:
I may be missing something here, but why can't the caller just perform the hashing if needed?

Does this change work if you want to sign an arbitrary message?

Reply:
How does the caller know when the hashing is needed?

The ledger impl. needs the raw tx data. This information is used to display the `type`, `amount` and `toAddress` to the user for confirmation. After confirmation it calculates the hash from the raw tx and signs it.

This is different from the file based impl. which directly accepts and signs the hash.

In both cases it's the hash that is signed, but extra metadata is needed for confirmation in case of the ledger impl.

Reply:
The old impl. was a low level function that signs the data. In case of the ledger impl. we need a method that is specific for signing a transaction. Therefore it's probably better to introduce a separate method `signTx` which accepts a tx and a generic one `signMsg` which just signs the data passed to it.

Reply:
Yeah I think a separate method for signing a transaction makes sense.

Reply:
Something like this?

``` dart
/// Signs an arbitrary message.
Future<List<int>> sign(List<int> message);

/// Signs the hash of the transaction.
Future<List<int>> signTx(AccountBlockTemplate tx);
```
With the impl. being.
``` dart
Future<List<int>> sign(List<int> message) async {
  return Crypto.sign(message, privateKey, (await getPublicKey()));
}

Future<List<int>> signTx(AccountBlockTemplate tx) async {
  return Crypto.sign(
      Crypto.digest(tx.hash.getBytes()!), privateKey, (await getPublicKey()));
}
```
This will give other implementations of the interface full control over the transaction information.

Reply:
Lgtm